packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -230,8 +232,14 @@ class WebReleaseBundle extends Target {
 
   @override
   Future<void> build(Environment environment) async {
+    final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
     for (final File outputFile in environment.buildDir.listSync(recursive: true).whereType<File>()) {
-      if (!globals.fs.path.basename(outputFile.path).contains('main.dart.js')) {
+      final String basename = globals.fs.path.basename(outputFile.path);
+      if (!basename.contains('main.dart.js')) {
+        continue;
+      }
+      // In release mode, do not output deps file or source map.
```

Comment:
Why not? It is common for one to need to debug a release build.

Reply:
They can still be debugged with run release though. Do you want sourcemaps in release builds?

Reply:
At least on our side we'd like to build the gallery on every commit and deploy it to something like Firebase hosting, so that if there's a bug we can quickly find the version where it started and debug it. So in that case it would be useful to have sourcemaps for all types of builds.

Reply:
interesting, I'll leave them in for now then

Reply:
Done